File: sdks/java/src/main/java/com/apollodeploy/signal/sdk/api/ContactPropertiesAPI.java
package com.apollodeploy.signal.sdk.api;

import com.fasterxml.jackson.core.type.TypeReference;
import com.apollodeploy.signal.sdk.models.Types.*;
import com.apollodeploy.signal.sdk.transport.Transport;
import java.time.Duration;
import java.util.List;

/** ContactPropertiesAPI API operations. */
public final class ContactPropertiesAPI {
    private final Transport transport;

    public ContactPropertiesAPI(Transport transport) {
        this.transport = transport;
    }

    public ContactPropertyPageResponse listContactProperties(
        String projectId
    ) {
        String path = "/v1/projects/${projectId}/contact-properties";
        path = path.replace("${projectId}", Transport.encodePathSegment(String.valueOf(projectId)));

        Transport.Request request = new Transport.Request("GET", path);
        return transport.execute(request, new TypeReference<ContactPropertyPageResponse>() {
        });
    }

    public ContactPropertyResponse getContactProperty(
        String projectId,
        String propertyId
    ) {
        String path = "/v1/projects/${projectId}/contact-properties/${propertyId}";
        path = path.replace("${projectId}", Transport.encodePathSegment(String.valueOf(projectId)));
        path = path.replace("${propertyId}", Transport.encodePathSegment(String.valueOf(propertyId)));

        Transport.Request request = new Transport.Request("GET", path);
        return transport.execute(request, new TypeReference<ContactPropertyResponse>() {
        });
    }

    public ContactPropertyResponse createContactProperty(
        String projectId,
        CreateContactPropertyBody body
    ) {
        String path = "/v1/projects/${projectId}/contact-properties";
        path = path.replace("${projectId}", Transport.encodePathSegment(String.valueOf(projectId)));

        Transport.Request request = new Transport.Request("POST", path);
        request.body(body, "application/json");
        return transport.execute(request, new TypeReference<ContactPropertyResponse>() {
        });
    }

    public ContactPropertyResponse updateContactProperty(
        String projectId,
        String propertyId,
        UpdateContactPropertyBody body
    ) {
        String path = "/v1/projects/${projectId}/contact-properties/${propertyId}";
        path = path.replace("${projectId}", Transport.encodePathSegment(String.valueOf(projectId)));
        path = path.replace("${propertyId}", Transport.encodePathSegment(String.valueOf(propertyId)));

        Transport.Request request = new Transport.Request("PATCH", path);
        request.body(body, "application/json");
        return transport.execute(request, new TypeReference<ContactPropertyResponse>() {
        });
    }

    public void deleteContactProperty(
        String projectId,
        String propertyId
    ) {
        String path = "/v1/projects/${projectId}/contact-properties/${propertyId}";
        path = path.replace("${projectId}", Transport.encodePathSegment(String.valueOf(projectId)));
        path = path.replace("${propertyId}", Transport.encodePathSegment(String.valueOf(propertyId)));

        Transport.Request request = new Transport.Request("DELETE", path);
        transport.executeVoid(request);
    }

}

File: sdks/kotlin/src/main/kotlin/com/apollodeploy/signal/sdk/api/ContactPropertiesAPI.kt
package com.apollodeploy.signal.sdk.api

import com.apollodeploy.signal.sdk.internal.Transport
import com.apollodeploy.signal.sdk.models.*
import io.ktor.http.HttpMethod
import io.ktor.http.encodeURLPathPart
import kotlinx.serialization.json.Json
import kotlinx.serialization.json.JsonObject
import kotlinx.serialization.json.encodeToJsonElement
import kotlinx.coroutines.flow.Flow

/** ContactPropertiesAPI API operations. */
class ContactPropertiesAPI(
    private val transport: Transport,
) {
/** listContactProperties operation. */
    suspend fun listContactProperties(
        projectId: String,
    ): ContactPropertyPageResponse {
        var path = "/v1/projects/\${projectId}/contact-properties"
        path = path.replace("\${projectId}", projectId.toString().encodeURLPathPart())

        val query = mutableListOf<Pair<String, String>>()
        val headers = mutableMapOf<String, String>()

        return transport.execute<ContactPropertyPageResponse>(
            method = HttpMethod.Get,
            path = path,
            query = query.ifEmpty { null },
            body = null,
            headers = headers.ifEmpty { null },
            operationTimeoutMs = null,
        )
    }

/** getContactProperty operation. */
    suspend fun getContactProperty(
        projectId: String,
        propertyId: String,
    ): ContactPropertyResponse {
        var path = "/v1/projects/\${projectId}/contact-properties/\${propertyId}"
        path = path.replace("\${projectId}", projectId.toString().encodeURLPathPart())
        path = path.replace("\${propertyId}", propertyId.toString().encodeURLPathPart())

        val query = mutableListOf<Pair<String, String>>()
        val headers = mutableMapOf<String, String>()

        return transport.execute<ContactPropertyResponse>(
            method = HttpMethod.Get,
            path = path,
            query = query.ifEmpty { null },
            body = null,
            headers = headers.ifEmpty { null },
            operationTimeoutMs = null,
        )
    }

/** createContactProperty operation. */
    suspend fun createContactProperty(
        projectId: String,
        body: CreateContactPropertyBody,
    ): ContactPropertyResponse {
        var path = "/v1/projects/\${projectId}/contact-properties"
        path = path.replace("\${projectId}", projectId.toString().encodeURLPathPart())

        val query = mutableListOf<Pair<String, String>>()
        val headers = mutableMapOf<String, String>()

        return transport.execute<ContactPropertyResponse>(
            method = HttpMethod.Post,
            path = path,
            query = query.ifEmpty { null },
            body = Json.encodeToJsonElement(body),
            contentType = "application/json",
            headers = headers.ifEmpty { null },
            operationTimeoutMs = null,
        )
    }

/** updateContactProperty operation. */
    suspend fun updateContactProperty(
        projectId: String,
        propertyId: String,
        body: UpdateContactPropertyBody,
    ): ContactPropertyResponse {
        var path = "/v1/projects/\${projectId}/contact-properties/\${propertyId}"
        path = path.replace("\${projectId}", projectId.toString().encodeURLPathPart())
        path = path.replace("\${propertyId}", propertyId.toString().encodeURLPathPart())

        val query = mutableListOf<Pair<String, String>>()
        val headers = mutableMapOf<String, String>()

        return transport.execute<ContactPropertyResponse>(
            method = HttpMethod.Patch,
            path = path,
            query = query.ifEmpty { null },
            body = Json.encodeToJsonElement(body),
            contentType = "application/json",
            headers = headers.ifEmpty { null },
            operationTimeoutMs = null,
        )
    }

/** deleteContactProperty operation. */
    suspend fun deleteContactProperty(
        projectId: String,
        propertyId: String,
    ): Unit {
        var path = "/v1/projects/\${projectId}/contact-properties/\${propertyId}"
        path = path.replace("\${projectId}", projectId.toString().encodeURLPathPart())
        path = path.replace("\${propertyId}", propertyId.toString().encodeURLPathPart())

        val query = mutableListOf<Pair<String, String>>()
        val headers = mutableMapOf<String, String>()

        transport.executeRaw(
            method = HttpMethod.Delete,
            path = path,
            query = query.ifEmpty { null },
            body = JsonObject(emptyMap()),
            headers = headers.ifEmpty { null },
            operationTimeoutMs = null,
        )
        return Unit
    }

}
